src/output-json-alert.c

```diff
@@ -173,6 +174,31 @@ static void AlertJsonSourceTarget(const Packet *p, const PacketAlert *pa,
     jb_close(js);
 }
 
+static void AlertJsonReference(
+        AlertJsonOutputCtx *json_output_ctx, const PacketAlert *pa, JsonBuilder *jb)
+{
+    if (!pa->s->references) {
+        return;
+    }
+
+    bool first = true;
+    const DetectReference *kv = pa->s->references;
+    while (kv) {
+        if (first) {
+            jb_open_array(jb, "references");
```

Comment:
Not needed to use `first` as you return early if `(!pa->s->references)`, right ?